packages/okx/src/index.ts

```diff
@@ -0,0 +1,44 @@
+import type { WalletInit, EIP1193Provider } from '@web3-onboard/common'
+import { createEIP1193Provider } from '@web3-onboard/common'
+
+import { CustomWindow } from './types.js'
+
+declare const window: CustomWindow
+
+function okx(): WalletInit {
+  if (typeof window === 'undefined') return () => null
+
+  return ({ device }) => {
+    return {
+      label: 'OKX Wallet',
+      getIcon: async () => (await import('./icon.js')).default,
+      getInterface: async () => {
+        const okxwalletInjectionExists = window.hasOwnProperty('okxwallet')
+
+        let provider: EIP1193Provider
+
+        if (okxwalletInjectionExists) {
+          provider = createEIP1193Provider(window['okxwallet'])
+        } else {
+          const downloadUrl =
+            device.type === 'mobile'
+              ? 'https://www.okx.com/download?deeplink=' +
+                encodeURIComponent(
+                  'okx://wallet/dapp/url?dappUrl=' +
+                    encodeURIComponent(window.location.href)
+                )
+              : 'https://www.okx.com/download'
+          window.open(downloadUrl, '_blank')
+          throw new Error('Please Install OKX Wallet before proceeding')
```

Comment:
```suggestion
          const downloadUrl =
            device.type === 'mobile'
              ? 'https://www.okx.com/download?deeplink=' +
                encodeURIComponent(
                  'okx://wallet/dapp/url?dappUrl=' +
                    encodeURIComponent(window.location.href)
                )
              : 'https://www.okx.com/download'
             throw new Error(
               createDownloadMessage('OKX Wallet', downloadUrl)
             )
```

`createDownloadMessage` is imported from web3-onboard/common
This provides a clean download link/message without routing the user without their consent

Reply:
👌 Okay, I'll get right on this.